libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp

```diff
@@ -827,6 +827,7 @@ extern "C" {
     uint16_t suites[cipher_cnt];
     memcpy_P(suites, cipher_list, cipher_cnt * sizeof(cipher_list[0]));
     br_ssl_client_zero(cc);
+    br_ssl_engine_add_flags(&cc->eng, BR_OPT_NO_RENEGOTIATION);  // forbid SSL renegociation, as we free the Private Key after handshake
```

Comment:
That's a good catch. I'm not quite sure whether it's the right solution or if we should undo the dropping of X509 after connection.

After looking at the RFC I see this could cause two problem in two ways (and dropping X509 would need to be undone):
1. When more 2^64 messages are sent and the sequence num needs to overflow.  This one we can probably ignore.
2. Servers are free to reject connections with this option sent in the HELLO.  Unfortunately I don't know how prevalent this would be.

I'm inclined to say no. 2 happens very infrequently and go with this fix.  If we get feedback otherwise we can undo this and the original drop-x509-after-handshake patch.